app/models/user/reputation_tokens/code_contribution_token.rb

```diff
@@ -5,6 +5,10 @@ class User::ReputationTokens::CodeContributionToken < User::ReputationToken
   levels %i[minor regular major]
   values({ minor: 5, regular: 10, major: 15 })
 
+  before_validation on: :create do
+    self.track = Track.find_by(repo_url: "https://github.com/#{repo}") unless track
```

Comment:
In theory I could also pass the repo URL from the webhook all the way up to this point, but I think the repo url is stable enough not to do that :)

Reply:
```suggestion
    self.track_id = Track.where(repo_url: "https://github.com/#{repo}").pluck(&:id).first unless track
```

This will save loading a whole model